nupic/research/frameworks/pytorch/imagenet/mixins/knowledge_distillation.py

```diff
@@ -0,0 +1,281 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import functools
+import sys
+import time
+
+import torch
+import torch.nn.functional as F
+from tqdm import tqdm
+
+class KnowledgeDistillation(object):
+    """
+    Sets the network to learn from a teacher model
+    """
+    def __init__(self):
```

Comment:
Make sure you update `self.execution_order`. (I'm guessing you were already planning to do this.)

Reply:
Please check the changes to execution_order, as we discussed